addons/account_check_printing/models/account_payment.py

```diff
@@ -123,7 +123,8 @@ def print_checks(self):
             }
         else:
             self.filtered(lambda r: r.state == 'draft').post()
-            self.write({'state': 'sent'})
+            if self.state != 'cancel':
+                self.write({'state': 'sent'})
```

Comment:
You can directly write like this: **self.filtered(lambda r: r.state != 'cancel').write({'state': 'sent'})**